pyccel/ast/utilities.py

```diff
@@ -541,22 +541,34 @@ def collect_loops(block, indices, new_index, language_has_vectors = False, resul
             current_level = new_level
 
         elif isinstance(line, Assign) and isinstance(line.lhs, IndexedElement) \
-                and isinstance(line.rhs, (PythonTuple, NumpyArray)) and not language_has_vectors:
-
+                and isinstance(line.rhs, (PythonTuple, NumpyArray)):
             lhs = line.lhs
             rhs = line.rhs
-            if isinstance(rhs, NumpyArray):
-                rhs = rhs.arg
-
-            lhs_rank = lhs.rank
-
-            new_assigns = [Assign(
-                            insert_index(expr=lhs,
-                                pos       = -lhs_rank,
-                                index_var = LiteralInteger(j)),
-                            rj) # lhs[j] = rhs[j]
-                          for j, rj in enumerate(rhs)]
-            collect_loops(new_assigns, indices, new_index, language_has_vectors, result = result)
+            if lhs.rank > rhs.rank:
+                for index in range(lhs.rank-rhs.rank):
+                    # If an index exists at the same depth, reuse it if not create one
+                    if index >= len(indices):
+                        indices.append(new_index(PythonNativeInt(),'i'))
+                    index_var = indices[index]
+                    lhs = insert_index(lhs, index, index_var)
+                collect_loops([Assign(lhs,rhs)], indices, new_index, language_has_vectors, result = result)
```

Comment:
The code is more readable if we follow a comma with an empty space. There are a couple missing here:
```suggestion
                        indices.append(new_index(PythonNativeInt(), 'i'))
                    index_var = indices[index]
                    lhs = insert_index(lhs, index, index_var)
                collect_loops([Assign(lhs, rhs)], indices, new_index, language_has_vectors, result = result)
```